src/SamplingSchemes.jl

```diff
@@ -0,0 +1,314 @@
+    max_add_steps::Int=1_000_000
+end
+
+"""
+    struct MixedMCRoutine <: MCRoutine
+
+A mutable struct representing a mixed Monte Carlo routine, where the main routine is used for the majority of the steps, 
+    and the backup routine is used when the main routine fails to accept a move. Currently, it is intended to use `MCRandomWalk` 
+    as the main routine and `MCDemonWalk` as the backup routine.
+
+# Fields
+- `main_routine::MCRoutine`: The main Monte Carlo routine.
+- `back_up_routine::MCRoutine`: The backup Monte Carlo routine.
+- `ns_params_main::NestedSamplingParameters`: The nested sampling parameters for the main routine.
+- `ns_params_back_up::NestedSamplingParameters`: The nested sampling parameters for the backup routine.
+"""
+@kwdef mutable struct MixedMCRoutine <: MCRoutine
+    main_routine::MCRoutine=MCRandomWalk()
+    back_up_routine::MCRoutine=MCDemonWalk()
+    ns_params_main::NestedSamplingParameters=NestedSamplingParameters(1000, 0.1, 0)
+    ns_params_back_up::NestedSamplingParameters=NestedSamplingParameters(10000, 0.01, 0)
+end
+
+
+
+
+"""
+    sort_by_energy!(liveset::LJAtomWalkers)
+
+Sorts the walkers in the liveset by their energy in descending order.
+
+# Arguments
+- `liveset::LJAtomWalkers`: The liveset of walkers to be sorted.
+
+# Returns
+- `liveset::LJAtomWalkers`: The sorted liveset.
+"""
+function sort_by_energy!(liveset::AtomWalkers)
+    sort!(liveset.walkers, by = x -> x.energy, rev=true)
+    # println("after sort ats[1].system_data.energy: ", ats[1].system_data.energy)
+    return liveset
+end
+
+"""
+    update_iter!(liveset::AtomWalkers)
+
+Update the iteration count for each walker in the liveset.
+
+# Arguments
+- `liveset::AtomWalkers`: The set of walkers to update.
+
+"""
+function update_iter!(liveset::AtomWalkers)
+    for at in liveset.walkers
+        at.iter += 1
+    end
+end
+
+
+"""
+    nested_sampling_step!(liveset::AtomWalkers, ns_params::NestedSamplingParameters, mc_routine::MCRandomWalk)
+
+Perform a single step of the nested sampling algorithm using the Monte Carlo random walk routine.
+
+Arguments
+- `liveset::AtomWalkers`: The set of atom walkers.
+- `ns_params::NestedSamplingParameters`: The parameters for nested sampling.
+- `mc_routine::MCRandomWalk`: The Monte Carlo random walk routine.
+
+Returns
+- `iter`: The iteration number after the step.
+- `emax`: The highest energy recorded during the step.
+- `liveset`: The updated set of atom walkers.
+- `ns_params`: The updated nested sampling parameters.
+"""
+function nested_sampling_step!(liveset::AtomWalkers, ns_params::NestedSamplingParameters, mc_routine::MCRandomWalk)
+    sort_by_energy!(liveset)
+    ats = liveset.walkers
+    lj = liveset.lj_potential
+    iter::Union{Missing,Int} = missing
+    emax::Union{Missing,typeof(0.0u"eV")} = liveset.walkers[1].energy
+    to_walk = ats[1]
+    accept, rate, at = MC_random_walk!(ns_params.mc_steps, to_walk, lj, ns_params.step_size, emax)
```

Comment:
Is `step_size` the maximum step size or the exact distance between a particle's initial and final positions?

Reply:
It's the maximum, a random number is drawn between `(-step_size, step_size)` uniformly. See [`single_atom_random_walk!()`](https://github.com/wexlergroup/FreeBird.jl/blob/0c5c0356045b0b87c5f6c9b41f52ded817de576e/src/AtomsMCMoves.jl#L96).